tests/frame/test_common.py

```diff
@@ -312,6 +313,41 @@ def test_join(df_raw: Any) -> None:
     compare_dicts(result_native, expected)
 
 
+@pytest.mark.parametrize(
+    "df_raw",
+    [
+        df_polars,
+        df_lazy,
+        df_pandas,
+        df_pandas_na,
+        df_lazy_na,
+        df_mpd,
+    ],
+)
+def test_cross_join(df_raw: Any) -> None:
+    df = nw.from_native(df_raw).select("a", "z")
+    result = df.join(df, how="cross")  # type: ignore[arg-type]
+
+    df_ = df.collect() if isinstance(df, nw.LazyFrame) else df
+
+    # List of dicts: [{"a": 1, "z": 7., ...}, {"a": 1, "z": 8., ...}, ...]
+    rows = [
+        {**left_row, **{k + "_right": v for k, v in right_row.items()}}
+        for left_row in df_.iter_rows(named=True)
+        for right_row in df_.iter_rows(named=True)
+    ]
+    # Pivot to single dict of values: {"a": [1, 1, ...], "z": [7., 8., ...], ...}
+    expected = defaultdict(list)
```

Comment:
could we also construct `expected` explicitly please? else we risk having a bug in `iter_rows` which hides this another bug in cross-join